### ISSUE TITLE
How to package avalonia program  into an  installation package  that can install on debian os and mac os.

### ISSUE THREAD

**Issue by fengyinyong** [AvaloniaUI/Avalonia#4313]
I'm working on the Avalonia program. But there are some difficulties in packaging installation package. When the program is published, I want to package it  into  an installation package that runs on a Debian OS and MAC OS. 
How to do it ??

**Comment by kekekeks** [comment]
For Linux you can use https://github.com/qmfrederik/dotnet-packaging

You'll need to add an icon to `/usr/share/icons` and a `.desktop` file to `/usr/share/applications`. 
Both can be added via
```
    <Content Include="myicon.png" CopyToPublishDirectory="PreserveNewest">
      <LinuxPath>/usr/share/icons/myicon.png</LinuxPath>
    </Content>
```

For OSX see https://avaloniaui.net/docs/packing/macOS

**Comment by fengyinyong** [comment]
Hi Nikita:
     Thanks for the solution you provided. According to the solution, I have solved my problem.And I also made a more detailed solution.When i am free,I will translate it to English and put in on the github.
I really appreciate your help。Best wishes to you
















At 2020-07-17 17:26:55, "Nikita Tsukanov" <notifications@github.com> wrote:

For Linux you can use https://github.com/qmfrederik/dotnet-packaging

You'll need to add an icon to /usr/share/icons and a .desktop file to /usr/share/applications.
Both can be added via

    <Content Include="myicon.png" CopyToPublishDirectory="PreserveNewest">
      <LinuxPath>/usr/share/icons/myicon.png</LinuxPath>
    </Content>


For OSX see https://avaloniaui.net/docs/packing/macOS

**Comment by fengyinyong** [comment]
Hi Nikita:
  Thanks for providing the solution that help me make deb install pakage last time.I really appreciate your help.
Now I hava two problems, I have read the avalonia docs,But I do not find rich text deidors control and webview control.
Therefore, Does avalonia support rich text editors and WebViews.Do you have any demos.
My Team are working on a project with Avalonia and hope that this avalonia will have a better future。
















At 2020-07-17 17:26:55, "Nikita Tsukanov" <notifications@github.com> wrote:

For Linux you can use https://github.com/qmfrederik/dotnet-packaging

You'll need to add an icon to /usr/share/icons and a .desktop file to /usr/share/applications.
Both can be added via

    <Content Include="myicon.png" CopyToPublishDirectory="PreserveNewest">
      <LinuxPath>/usr/share/icons/myicon.png</LinuxPath>
    </Content>


For OSX see https://avaloniaui.net/docs/packing/macOS

**Comment by kekekeks** [comment]
WebView - https://github.com/CefNet/CefNet
Rich editor - https://github.com/AvaloniaUI/AvaloniaEdit/​, proper Flow Document model will be available in 0.11
